storage/tests/unit/test_bucket.py

```diff
@@ -569,6 +584,21 @@ def test_get_blob_hit_w_user_project(self):
         self.assertEqual(kw["path"], "/b/%s/o/%s" % (NAME, BLOB_NAME))
         self.assertEqual(kw["query_params"], {"userProject": USER_PROJECT})
 
+    def test_get_blob_hit_w_generation(self):
+        NAME = "name"
+        BLOB_NAME = "blob-name"
+        GENERATION = 1512565576797178
+        connection = _Connection({"name": BLOB_NAME})
```

Comment:
For clarity, add the generation to the properties here, and then assert that it is read in correctly below:

```python
        connection = _Connection({"name": BLOB_NAME, generation=GENERATION})
```